eqctl/eqctl.c

```diff
@@ -0,0 +1,219 @@
+{
+	FILE *fh;
+	int n = 0;
+	int n_max = smax / sizeof(unsigned int);
+	int separator;
+
+	fh = fopen(setup, "r");
+	while (fscanf(fh, "%u", &data[n]) != EOF && n < n_max) {
+		if (n > 0)
+			printf(",");
+		printf("%u", data[n]);
+		separator = fgetc(fh);
+		//printf("<%c>", (char)separator);
+		while (separator != ',' && separator != EOF) {
+			separator = fgetc(fh);
+			//printf("<%c>", (char)separator);
+		}
+		n++;
+	}
+	fclose(fh);
+	printf("\n");
+	return n;
+}
+
+int main(int argc, char *argv[])
+{
+	snd_ctl_t *ctl;
+	snd_ctl_elem_id_t *id;
+	snd_ctl_elem_info_t *info;
+	snd_ctl_elem_value_t *value;
+	uint32_t *config;
+	unsigned int *user_data;
+	unsigned char *bytes;
+	char nname[256];
+	int ret;
+	int ctrl_size;
+	int read;
+	int write;
+	int type;
+	int i;
+	char opt;
+	int n;
+	int buffer_size;
+	int mode = SND_CTL_NONBLOCK;
+	char *dev = "hw:0";
+	char *cname = NULL;
+	char *setup = NULL;
+	int set = 0;
+
+	while ((opt = getopt(argc, argv, "hD:c:s:n:")) != -1) {
+		switch (opt) {
+		case 'D':
+			dev = optarg;
+			break;
+		case 'c':
+			cname = optarg;
+			break;
+		case 'n':
+			sprintf(nname, "numid=%d", atoi(optarg));
+			cname = nname;
+			break;
+		case 's':
+			setup = optarg;
+			set = 1;
+			break;
+		case 'h':
+			usage(argv[0]);
+			exit(EXIT_SUCCESS);
+			break;
+		default:
+			usage(argv[0]);
+			exit(EXIT_FAILURE);
+		}
+	}
+
+	/* The control need to be defined. */
+	if (!cname) {
+		fprintf(stderr, "Error: No control was requested.\n");
+		usage(argv[0]);
+		exit(EXIT_FAILURE);
+
+	}
+
+	/* Open the device, mixer control and get read/write/type properties.
+	 */
+	ret = snd_ctl_open(&ctl, dev, mode);
+	if (ret) {
+		fprintf(stderr, "Error: Could not open device %s.\n", dev);
+		exit(EXIT_FAILURE);
+	}
+
+	snd_ctl_elem_info_alloca(&info);
+	snd_ctl_elem_id_alloca(&id);
+	snd_ctl_elem_value_alloca(&value);
+	ret = snd_ctl_ascii_elem_id_parse(id, cname);
+	if (ret) {
+		fprintf(stderr, "Error: Can't find %s.\n", cname);
+		exit(EXIT_FAILURE);
+	}
+
+	snd_ctl_elem_info_set_id(info, id);
+	ret = snd_ctl_elem_info(ctl, info);
+	if (ret) {
+		fprintf(stderr, "Error: Could not get elem info.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	ctrl_size = snd_ctl_elem_info_get_count(info);
+	read = snd_ctl_elem_info_is_tlv_readable(info);
+	write = snd_ctl_elem_info_is_tlv_writable(info);
+	type = snd_ctl_elem_info_get_type(info);
+	if (!read || !write || type != SND_CTL_ELEM_TYPE_BYTES) {
+		fprintf(stderr, "Error: No read, write or bytes capability.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	/* Next allocate buffer for tlv write/read. The buffer needs a two
+	 * words header with tag (SOF_CTRL_CMD_BINARY) and size in bytes.
+	 */
+	buffer_size = ctrl_size + 2 * sizeof(unsigned int);
+	user_data = calloc(1, buffer_size);
+	user_data[0] = SOF_CTRL_CMD_BINARY;
+	if (set) {
+		fprintf(stderr, "Applying configuration \"%s\" ", setup);
+		fprintf(stderr, "into device %s control %s.\n", dev, cname);
```

Comment:
stdout

Reply:
OK, I'll change. This was a trick to get coefficients back to a correct syntax file without other information.

Reply:
But better to add later file output with e.g. -o if need for convenience. I'm not also totally sure about the ASCII file format for numbers data but use this for now.